src/js/signin/containers/Main.jsx

```diff
@@ -0,0 +1,167 @@
+      this.getUserData(this.props.dispatch);
+      this.getLogoutUrl();
+    }
+  }
+
+  getLogoutUrl() {
+    this.logoutUrlRequest = fetch(`${environment.API_URL}/v0/sessions`, {
+      method: 'DELETE',
+      headers: new Headers({
+        Authorization: `Token token=${sessionStorage.userToken}`
+      })
+    }).then(response => {
+      return response.json();
+    }).then(json => {
+      this.props.onUpdateLogoutUrl(json.logout_via_get);
+    });
+  }
+
+  handleLogin() {
+    this.loginUrlRequest = handleLogin(this.props.login.loginUrl, this.props.onUpdateLoginUrl);
+  }
+
+  handleSignup() {
+    window.dataLayer.push({ event: 'register-link-clicked' });
+    const myLoginUrl = this.props.login.loginUrl;
+    if (myLoginUrl) {
+      window.dataLayer.push({ event: 'register-link-opened' });
+      const receiver = window.open(`${myLoginUrl}&op=signup`, '_blank', 'resizable=yes,scrollbars=1,top=50,left=500,width=500,height=750');
+      receiver.focus();
+    }
+  }
+
+  handleLogout() {
+    window.dataLayer.push({ event: 'logout-link-clicked' });
+    const myLogoutUrl = this.props.login.logoutUrl;
+    if (myLogoutUrl) {
+      window.dataLayer.push({ event: 'logout-link-opened' });
+      const receiver = window.open(myLogoutUrl, '_blank', 'resizable=yes,scrollbars=1,top=50,left=500,width=500,height=750');
+      receiver.focus();
+    }
+  }
+
+  checkTokenStatus() {
+    if (sessionStorage.userToken) {
+      if (moment() > moment(sessionStorage.entryTime).add(45, 'm')) {
+        // TODO(crew): make more customized prompt.
+        if (confirm('For security, you’ll be automatically signed out in 2 minutes. To stay signed in, click OK.')) {
+          this.handleLogin();
+        } else {
+          this.handleLogout();
+        }
+      } else {
+        if (this.getUserData(this.props.dispatch)) {
+          this.props.onUpdateLoggedInStatus(true);
+        }
+      }
+    } else {
+      this.props.onUpdateLoggedInStatus(false);
+    }
+  }
+
+  render() {
+
+    if (this.props.verify) {
+      return (
+        <Signin
+          handleSignup={this.handleSignup}
+          handleLogin={this.handleLogin}/>
+      );
+    }
+
+    return (
+      <Signin
+        currentlyLoggedIn={this.props.login.currentlyLoggedIn}
+        handleSignup={this.handleSignup}
+        handleLogin={this.handleLogin}/>
+    );
+  }
+}
+
+const mapStateToProps = (state) => {
+  const userState = state.user;
+  return {
+    login: userState.login,
+    profile: userState.profile
+  };
+};
+
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    onUpdateLoginUrl: (update) => {
+      dispatch(updateLogInUrl(update));
+    },
+    onUpdateVerifyUrl: (update) => {
+      dispatch(updateVerifyUrl(update));
+    },
+    onUpdateLogoutUrl: (update) => {
+      dispatch(updateLogoutUrl(update));
+    },
+    onUpdateLoggedInStatus: (update) => {
+      dispatch(updateLoggedInStatus(update));
+    },
+    dispatch
+  };
```

Comment:
Not a fan of needlessly renaming all the actions here either. Would rather just map them directly to the same names. Also, mapping `dispatch` is redundant since it should already be available in props, I believe...